Utils.Net/IO/FileUtils.cs

```diff
@@ -0,0 +1,320 @@
+            while (firstReadBytes == bufferSize);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Delete a file. Do not throw an exception if the file does not exist.
+        /// Offer the possibility to delete read-only files.
+        /// </summary>
+        /// <param name="filePath">Path of the file to delete.</param>
+        /// <param name="force">true if you want to delete the file even when it is read-only or hidden.</param>
+        public static void DeleteFile(string filePath, bool force = true)
+        {
+            if (!File.Exists(filePath))
+                return;
+
+            if (force)
+            {
+                RemoveReadOnly(filePath);
+                RemoveHidden(filePath);
+            }
+
+            File.Delete(filePath);
+        }
+
+        /// <summary>
+        /// Copy a file by replacing the existing one.
+        /// </summary>
+        /// <param name="sourcePath">Path of the file to copy.</param>
+        /// <param name="destPath">Path of the file to create.</param>
+        /// <param name="createFolder">If true, the folder of <see cref="destPath"/> 
+        /// will be created if necessary.</param>
+        /// <param name="removeReadOnly">Remove the read-only flag on the newly
+        /// created file. Also remove the read-only flag on the destination file
+        /// before the copy to make sure the operation succeeds.</param>
+        /// <param name="removeHidden">Remove the hidden flag on the newly
+        /// created file. Also remove the hidden flag on the destination file
+        /// before the copy to make sure the operation succeeds.</param>
+        public static void CopyFile(
+            string sourcePath,
+            string destPath,
+            bool createFolder = false,
+            bool removeReadOnly = false,
+            bool removeHidden = false)
+        {
+            if (!File.Exists(sourcePath))
+                return;
+
+            if (createFolder)
+            {
+                string dir = Path.GetDirectoryName(destPath);
+                if (!string.IsNullOrEmpty(dir))
+                    Directory.CreateDirectory(dir);
+            }
+
+            if (removeReadOnly)
+            {
+                RemoveReadOnly(destPath);
+            }
+
+            if (removeHidden)
+            {
+                RemoveHidden(destPath);
+            }
+
+            File.Copy(sourcePath, destPath, true);
+
+            if (removeReadOnly)
+            {
+                RemoveReadOnly(destPath);
+            }
+
+            if (removeHidden)
+            {
+                RemoveHidden(destPath);
+            }
+        }
+
+        #endregion
+
+        #region ReadOnly attribute
+        /// <summary>
+        /// Check if file is read-only
+        /// </summary>
+        public static bool IsReadOnly(string filePath)
+        {
+            return (File.GetAttributes(filePath) & FileAttributes.ReadOnly) == FileAttributes.ReadOnly;
```

Comment:
It will be nice to create generic functions to check/set/remove attribute to reduce duplicated code. And define specific functions for ReadOnly and Hidden which execute the generic one, if you want to simplify the calls.